web/controllers/contact_controller.ex

```diff
@@ -0,0 +1,58 @@
+defmodule ResearchResource.ContactController do
+  use ResearchResource.Web, :controller
+
+  @contact_email Application.get_env(:research_resource, :contact_email)
+
+  def index(conn, _params) do
+    render conn, "index.html"
+  end
+
+  def create(conn,
+    %{ "callback" => %{"name" => name, "phone" => phone} = callback }
+  ) when name != "" and phone != "" do
+    conn
+    |> send_callback_email(callback)
+    |> put_flash(:info, "Thank you for requesting a call back, one of our staff will be in touch soon.")
+    |> redirect(to: contact_path(conn, :index))
+  end
+  def create(conn,
+    %{ "message" => %{"name" => name, "message" => message} = details }
+  ) when name != "" and message != "" do
+    conn
+    |> send_message_email(details)
+    |> put_flash(:info, "Thank you for your message, one of our staff will be in touch soon.")
+    |> redirect(to: contact_path(conn, :index))
+  end
```

Comment:
I may have created a generic function for these and passed in message and phone or something, but again just personal preference, looks good 👍